src/root.tsx

```diff
@@ -1,37 +1,39 @@
 // @refresh reload
+import "./code.css";
+import "virtual:windi.css";
+
+import { Stylesheet } from "@solidjs/meta";
+import { Suspense, useContext } from "solid-js";
+import { isServer } from "solid-js/web";
+import { MDXProvider } from "solid-mdx";
 import {
-  Routes,
-  Meta,
-  Link,
+  Body,
   FileRoutes,
-  Scripts,
-  Html,
   Head,
-  Body,
+  Html,
+  Link,
+  Meta,
+  Routes,
+  Scripts,
 } from "solid-start";
+import { useLocation } from "solid-start";
+import { HttpHeader, ServerContext } from "solid-start/server";
 import { parseCookie } from "solid-start/session";
-import "./code.css";
-import "virtual:windi.css";
 
 import {
   Config,
   ConfigProvider,
   defaultConfig,
 } from "./components/context/ConfigContext";
 import { PageStateProvider } from "./components/context/PageStateContext";
-
-import { MDXProvider } from "solid-mdx";
+import { Main } from "./components/Main";
 import Nav from "./components/nav/Nav";
 import md from "./md";
-import { createRenderEffect, Suspense, useContext } from "solid-js";
-import { Main } from "./components/Main";
-import { HttpHeader, ServerContext } from "solid-start/server";
-import { isServer } from "solid-js/web";
-import { useLocation } from "solid-start";
-import { Stylesheet } from "@solidjs/meta";
 
 function useCookies() {
   const context = useContext(ServerContext);
+  // TODO: isServer is typed as true, not boolean, in solid-js/web. Is that right?
+  // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
   const cookies = isServer
```

Comment:
@Jutanium I suspect this is either an unnecessary ternary or hitting a real issue other users are going to hit from [`@typescript-eslint/no-unnecessary-condition`](https://typescript-eslint.io/rules/no-unnecessary-condition). If `isServer` is typed as `true` instead of `boolean`, lint rules that check for unnecessary conditions are going to consider logic on it unnecessary.

In other words, because `isServer` is declared as `true`, the lint rule thinks the false condition in the ternary is unreachable code.

Reply:
@nksaraf @ryansolid any idea why it’s typed as `true?`

Reply:
I think it’s for tree shaking reasons